drivers/sensor/tdk/icm456xx/icm45686.c

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2025 TDK Invensense
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include "icm45686.h"
+
+#include <zephyr/drivers/sensor/icm456xx.h>
+#include <zephyr/sys/byteorder.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(ICM45686, CONFIG_SENSOR_LOG_LEVEL);
+
+static uint32_t convert_gyr_fs_to_enum(uint32_t val)
+{
+	uint32_t fs_enum = 0;
+
+	if (val < 500 && val >= 250) {
+		fs_enum = GYRO_CONFIG0_GYRO_UI_FS_SEL_250_DPS;
+	} else if (val < 1000 && val >= 500) {
+		fs_enum = GYRO_CONFIG0_GYRO_UI_FS_SEL_500_DPS;
+	} else if (val < 2000 && val >= 1000) {
+		fs_enum = GYRO_CONFIG0_GYRO_UI_FS_SEL_1000_DPS;
+	} else if (val == 2000) {
+		fs_enum = GYRO_CONFIG0_GYRO_UI_FS_SEL_2000_DPS;
+	}
+	return fs_enum;
+}
+
```

Comment:
We can also expose how to set odr and fs below using the driver APIs:
inv_imu_set_gyro_frequency()
inv_imu_set_gyro_fsr()

Reply:
icm45686 files are removed